src/deck/drivers/src/audio_deck.c

```diff
@@ -120,80 +113,16 @@ static uint8_t spi_tx_buffer[SPI_N_BYTES];
 static uint8_t spi_rx_buffer[SPI_N_BYTES];
 static uint8_t temp_spi_rx_buffer[SPI_N_BYTES];
 
-int pin_state = 0;
-
-////////////////////////////////////////// DEBUGGING  ///////////////////////////////////////
-uint8_t tx_counter = 0;
-
 /////////////////////////////// AUDIO DECK FUNCTIONS  ///////////////////////////////////////
 
-void spiReadWrite(uint8_t *data_read, const uint8_t *data_write, size_t data_length) {
-	spiBeginTransaction(spi_speed);
-	// Removing the temporary buffers below increased the publishing rate.
-	//memcpy(temp_spi_tx_buffer, data_write, data_length);
-	//uint8_t success = spiExchange(data_length, temp_spi_tx_buffer,
-	//		temp_spi_rx_buffer);
-	spiExchange(data_length, data_write, data_read);
-	spiEndTransaction();
-}
-
-// TODO(FD) below functions are not really needed anymore, but we keep them
-// for debugging.
-
-static uint8_t temp_spi_tx_buffer[SPI_N_BYTES];
-
-
-static uint8_t spiRead(void *data_read, size_t data_length) {
-	spiBeginTransaction(spi_speed);
-	memset(temp_spi_tx_buffer, 0, data_length);
-	uint8_t success = spiExchange(data_length, temp_spi_tx_buffer,
-			temp_spi_rx_buffer);
-	spiEndTransaction();
-	memcpy(data_read, temp_spi_rx_buffer, data_length);
-	return success;
-}
-
-static uint8_t spiWrite(const void *data_write,
-		size_t data_length) {
-	spiBeginTransaction(spi_speed);
-	memcpy(temp_spi_tx_buffer, data_write, data_length);
-	uint8_t success = spiExchange(data_length, temp_spi_tx_buffer,
-			temp_spi_rx_buffer);
-	spiEndTransaction();
-	return success;
-}
-
-
-/**
- *  Fill frequency data after the audio data in the CRTP packet.
- */
-void fill_packet_data_fbins(uint8_t packet_data[], uint8_t packet_count, uint8_t packet_size) {
-	for (uint8_t i = 0; i < packet_size; i++) {
-#ifndef DEBUG_SPI
-		packet_data[i] = crtp_tx_buffer[AUDIO_N_BYTES + packet_count * CRTP_MAX_PAYLOAD + i];
-#endif
-	}
-	//memcpy(packet_data, &crtp_tx_buffer[AUDIO_N_BYTES + packet_count * CRTP_MAX_PAYLOAD], packet_size);
```

Comment:
This function and fill_packet_data_audio, were really just copying buffers from one place to the other, so I replaced them alltogether with "memcpy".